ouroboros-consensus-diffusion/app/conformance-test-runner/Main.hs

```diff
@@ -23,9 +28,27 @@ import Ouroboros.Network.OrphanInstances ()
 import Ouroboros.Network.PeerSelection (PeerAdvertise (..), PortNumber)
 import Ouroboros.Network.PeerSelection.LedgerPeers
 import Ouroboros.Network.PeerSelection.State.LocalRootPeers (HotValency (..), WarmValency (..))
+import System.Exit (ExitCode (..))
 import Test.Consensus.PointSchedule
 import Test.Consensus.PointSchedule.Peers (PeerId (..), Peers (Peers), getPeerIds)
 
+data ExitStatus = InternalError | BadUsage | Flags (Set StatusFlag)
+
+data StatusFlag = TestFailed | ContinueShrinking deriving (Eq, Ord)
+
+exitStatusToCode :: ExitStatus -> ExitCode
+exitStatusToCode = \case
+  InternalError -> ExitFailure (bit 0)
+  BadUsage -> ExitFailure (bit 1)
+  Flags flags ->
+    if null flags
+      then ExitSuccess
+      else ExitFailure $ foldr ((.|.) . flagToCode) 0 flags
```

Comment:
clearer to express via a monoid:

```suggestion
      else ExitFailure $ getIOr $ foldMap flagToCode flags
```

and change the type of `flagToCode` to be `IOr Int`